app/src/main/java/com/p2p/presentation/tuttifrutti/review/TuttiFruttiReviewFragment.kt

```diff
@@ -1,42 +1,81 @@
+    override val viewModel: TuttiFruttiReviewViewModel by viewModels()
+
+    override val gameInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentReviewTuttiFruttiBinding =
+        FragmentReviewTuttiFruttiBinding::inflate
+
+    private lateinit var tuttiFruttiReviewRoundAdapter: TuttiFruttiReviewRoundAdapter
+
+
+    override fun initValues() {
+        viewModel.initializeBaseRoundPoints(
+            gameViewModel.actualRound.value!!,
+            gameViewModel.finishedRoundInfos.value!!
+        )
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-
-        // Delete this! It's awful and we should neve do this.
-        activity?.findViewById<View>(R.id.activity_progress_overlay)?.isVisible = false
-        viewmodel.finishedRoundInfos.observe(viewLifecycleOwner) { finishedRoundInfo ->
-            MaterialAlertDialogBuilder(requireContext())
-                .setMessage(finishedRoundInfo.joinToString("\n\n") {
-                    "${it.player} ${if (it.saidEnough) "(dijo basta)" else ""} --> ${it.categoriesWords}"
-                })
-                .show()
+    override fun initUI() {
+        super.initUI()
+        gameViewModel.stopLoading()
+        setupReviewCategoriesRecycler()
+        gameBinding.finishReviewButton.setOnClickListener { viewModel.sendRoundPoints() }
+    }
+
+    private fun setupReviewCategoriesRecycler() = with(gameBinding.reviewCategoriesRecycler) {
+        layoutManager = LinearLayoutManager(context)
+        adapter = TuttiFruttiReviewRoundAdapter(viewModel::onChangeRoundPoints).also {
+            this@TuttiFruttiReviewFragment.tuttiFruttiReviewRoundAdapter = it
         }
     }
 
+    override fun setupObservers() {
+        with(gameViewModel) {
+            actualRound.observe(viewLifecycleOwner) {
+                gameBinding.actualRound.text = resources.getString(R.string.tf_actual_round, it.number)
+                gameBinding.totalRounds.text = resources.getString(R.string.tf_total_rounds, totalRounds.value)
+                gameBinding.letter.text = resources.getString(R.string.tf_letter, it.letter)
+            }
+            finishedRoundInfos.observe(viewLifecycleOwner) { finishedRoundInfo ->
+                tuttiFruttiReviewRoundAdapter.finishedRoundInfo = finishedRoundInfo
+                gameBinding.enoughPlayer.text =
+                    resources.getString(R.string.tf_enough_player, finishedRoundInfo.find { it.saidEnough }!!.player)
+            }
+        }
+        with(viewModel) {
+            finishedRoundPointsInfo.observe(viewLifecycleOwner) {
+                tuttiFruttiReviewRoundAdapter.finishedRoundPointsInfo = it
+            }
+        }
+
+        super.setupObservers()
+    }
+
+    override fun onEvent(event: TuttiFruttiReviewEvents) = when (event) {
+        FinishRoundReview -> gameViewModel.setFinishedRoundPointsInfos(viewModel.finishedRoundPointsInfo.value!!)
```

Comment:
El evento `FinishRoundReview` debería pasar el `finishedRoundPointsInfo`, no se debería agarrar desde el viewmodel. Onda
 ```suggestion
        is FinishRoundReview -> gameViewModel.setFinishedRoundPointsInfos(event.finishedRoundPointsInfo)
```

Reply:
Genial!